lib/ruby-debugger.coffee

```diff
@@ -29,3 +31,12 @@ class RubyDebugger
       @panel.hide()
     else
       @panel.show()
+
+  addBreakpoint: ->
+    if @context.isConnected()
+      editor = atom.workspace.getActiveTextEditor()
+      path = editor.getPath()
+      lineNumber = editor.getSelectedBufferRange().start.row + 1
+      @context.client.addBreakpoint path, lineNumber
```

Comment:
I suggest you add a `addBreakpoint` method to DebuggerContext instead that delegates and adds the resolved Promise's new breakpoint to it's list of breakpoints (so a `@breakpoints` variable in `DebuggerContext`)

If you do that you can remove the event emitted `breakpointAdded` as well.